File: RepoTransBench/target_projects/Python/C++/mardix_flask-recaptcha/tests/original/test_flask_recaptcha_extra.cpp
#include <gtest/gtest.h>
#include "flask_recaptcha.h"

using namespace mardix;

TEST(ReCaptchaExtra, ParamsInit) {
    ReCaptcha r("A", "B", "light", "image", "compact", "en", 1, true);
    EXPECT_EQ(r.site_key(), "A");
    EXPECT_EQ(r.secret_key(), "B");
    EXPECT_EQ(r.theme(), "light");
    EXPECT_EQ(r.type(), "image");
    EXPECT_EQ(r.size(), "compact");
    EXPECT_EQ(r.language(), "en");
    EXPECT_EQ(r.tabindex(), 1);
}

TEST(ReCaptchaExtra, EnabledProperty) {
    ReCaptcha r("X", "Y", "light", "image", "normal", "en", 1, true);
    EXPECT_TRUE(r.is_enabled());
    ReCaptcha r2("X", "Y", "light", "image", "normal", "en", 1, false);
    EXPECT_FALSE(r2.is_enabled());
}

TEST(ReCaptchaExtra, GetCodeReturnsString) {
    ReCaptcha r("something", "else");
    std::string code = r.get_code();
    EXPECT_FALSE(code.empty());
}

TEST(ReCaptchaExtra, VerifyEmptyResponseToken) {
    ReCaptcha r("a", "b", true);
    EXPECT_FALSE(r.verify("", "host"));
}

TEST(ReCaptchaExtra, InitAppWithMinimum) {
    // Simulate only with site_key and secret_key
    struct App {
        std::map<std::string, std::string> config;
        std::function<std::map<std::string, std::string>()> _proc;
        App() {
            config = {
                {"RECAPTCHA_SITE_KEY", "k"},
                {"RECAPTCHA_SECRET_KEY", "s"}
            };
        }
        void context_processor(std::function<std::map<std::string, std::string>()> fn) {
            _proc = fn;
        }
    };
    App app;
    ReCaptcha r(app.config["RECAPTCHA_SITE_KEY"], app.config["RECAPTCHA_SECRET_KEY"]);
    app.context_processor([&]() {
        std::map<std::string, std::string> c;
        c["recaptcha"] = r.get_code();
        return c;
    });
    auto c = app._proc();
    EXPECT_TRUE(c.find("recaptcha") != c.end());
    EXPECT_FALSE(c["recaptcha"].empty());
}

File: RepoTransBench/source_projects/Python/mardix_flask-recaptcha/test_flask_recaptcha_extra.py
import pytest
from flask_recaptcha import ReCaptcha

def test_params_init():
    r = ReCaptcha(site_key="A", secret_key="B", theme="light", type="image", size="compact", language="en", tabindex=1)
    assert r.site_key == "A"
    assert r.secret_key == "B"
    assert r.theme == "light"
    assert r.type == "image"
    assert r.size == "compact"
    assert r.language == "en"
    assert r.tabindex == 1

def test_enabled_property():
    r = ReCaptcha(is_enabled=True)
    # Some implementations use attribute and property; this covers both possibilities
    assert getattr(r, "is_enabled") is True or getattr(r, "enabled", True) is True
    r2 = ReCaptcha(is_enabled=False)
    assert getattr(r2, "is_enabled") is False or getattr(r2, "enabled", False) is False

def test_get_code_returns_string():
    r = ReCaptcha(site_key="something", secret_key="else")
    code = r.get_code()
    assert isinstance(code, str)

def test_verify_empty_response_token(monkeypatch):
    import sys
    import flask_recaptcha
    class DummyRequests:
        @staticmethod
        def get(*a, **k):
            class Resp:
                status_code = 200
                def json(self): return {"success": True}
            return Resp()
    class DummyRequest:
        form = {}
        environ = {"REMOTE_ADDR": "host"}
    sys.modules["requests"] = DummyRequests
    sys.modules["flask_recaptcha.requests"] = DummyRequests
    monkeypatch.setattr(flask_recaptcha, "request", DummyRequest)
    r = ReCaptcha(site_key="a", secret_key="b", is_enabled=True)
    # Empty response token returns False
    assert r.verify("", "host") is False

def test_init_app_with_minimum(monkeypatch):
    import sys
    import flask_recaptcha
    # Patch Markup to prevent NameError on import
    class FakeMarkup(str): pass
    sys.modules["Markup"] = FakeMarkup
    monkeypatch.setitem(flask_recaptcha.__dict__, "Markup", FakeMarkup)
    class App:
        def __init__(self):
            self.config = {
                "RECAPTCHA_SITE_KEY": "k",
                "RECAPTCHA_SECRET_KEY": "s",
            }
            self._proc = None
        def context_processor(self, fn):
            self._proc = fn
            return fn
    app = App()
    r = ReCaptcha()
    r.init_app(app)
    c = app._proc()
    assert "recaptcha" in c and isinstance(c["recaptcha"], str)